src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachExceptionRecordToExistingCaseTest.java

```diff
@@ -293,71 +467,168 @@ void should_create_error_if_type_in_incorrect() {
+            submittedScannedRecords(),
+            "$.data.scannedDocuments[1].fileName",
+            WireMock.equalTo(EXCEPTION_RECORD_FILENAME)
+        );
+        verifyRequestPattern(
+            submittedScannedRecords(),
+            "$.event.summary",
+            WireMock.equalTo(String.format(
+                "Attaching exception record(%d) document numbers:[%s] to case:%s",
+                EXCEPTION_RECORD_ID,
+                EXCEPTION_RECORD_DOCUMENT_NUMBER,
+                CASE_REF
+            ))
+        );
+        verifyRequestPattern(
+            submittedScannedRecords(),
+            "$.event.id",
+            WireMock.equalTo(EVENT_ID)
+        );
+        verifyRequestPattern(
+            submittedScannedRecords(),
+            "$.event_token",
+            WireMock.equalTo(EVENT_TOKEN)
+        );
     }
 
-    private MappingBuilder ccdGetCaseMapping() {
-        return get("/cases/" + CASE_REF)
-            .withHeader(AUTHORIZATION, containing(MOCKED_IDAM_TOKEN_SIG))
-            .withHeader("ServiceAuthorization", containing(MOCKED_S2S_TOKEN_SIG));
+    private CaseDetails.CaseDetailsBuilder exceptionRecordBuilder(String attachToCaseReference) {
+        return exceptionRecordBuilder(attachToCaseReference, null, null, CASE_TYPE_EXCEPTION_RECORD);
     }
 
-    private CaseDetails.CaseDetailsBuilder exceptionRecordBuilder() {
+    private CaseDetails.CaseDetailsBuilder exceptionRecordBuilder(
+        String attachToCaseReference,
+        String searchCaseReferenceType,
+        String searchCaseReference,
+        String caseTypeId
+    ) {
         return CaseDetails.builder()
             .jurisdiction(JURISDICTION)
             .id(EXCEPTION_RECORD_ID)
-            .caseTypeId("ExceptionRecord")
-            .data(exceptionDataWithDoc(SCANNED_RECORD, CASE_REF));
+            .caseTypeId(caseTypeId)
+            .data(
+                exceptionDataWithDoc(
+                    SCANNED_RECORD,
+                    attachToCaseReference,
+                    searchCaseReferenceType,
+                    searchCaseReference
+                )
+            );
+    }
+
+    private void mockCcdExceptionRecordSearch(
+        String exceptionRecordCcdId,
+        ResponseDefinitionBuilder responseBuilder
+    ) {
+        givenThat(
+            get("/cases/" + exceptionRecordCcdId)
+                .withHeader(AUTHORIZATION, containing(MOCKED_IDAM_TOKEN_SIG))
+                .withHeader(SERVICE_AUTHORIZATION_HEADER, containing(MOCKED_S2S_TOKEN_SIG))
+                .willReturn(responseBuilder)
+        );
+    }
+
+    private void mockCaseSearchByCcdId(String ccdId, ResponseDefinitionBuilder responseBuilder) {
+        givenThat(
+            get("/cases/" + ccdId)
+                .withHeader(AUTHORIZATION, containing(MOCKED_IDAM_TOKEN_SIG))
+                .withHeader(SERVICE_AUTHORIZATION_HEADER, containing(MOCKED_S2S_TOKEN_SIG))
+                .willReturn(responseBuilder)
+        );
+    }
```

Comment:
`mockCcdExceptionRecordSearch` and `mockCaseSearchByCcdId` methods are identical

Reply:
refactoring gone bad

Reply:
removed `mockCcdExceptionRecordSearch`